### PR TITLE
[docs] Add release notes (draft/in progress) for 0.2

### PR DESCRIPTION
**Pull request by spouliot**

### CODE REVIEW

File: docs/releases/Embeddinator-4000_0.2.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+* macOS 10.12 (Sierra) or later;
+* Xcode 8.3.2 or later;
+* Mono 5.0 or later;
+
+Optional
+--------
+
+* [Visual Studio for Mac](https://www.visualstudio.com/vs/visual-studio-mac/);
+* [Xamarin.iOS 10.11.0.136 or later](https://jenkins.mono-project.com/view/Xamarin.MaciOS/job/xamarin-macios-builds-master/) **Links to preview builds**. Required for iOS, tvOS and watchOS support;
```
Comment (rolfbjarne) [inline docs/releases/Embeddinator-4000_0.2.md]: Updating the version number here every time it's also updated in Make.config.inc seems redundant and tedious, maybe we can update the version number when the release is branched instead?

File: docs/releases/Embeddinator-4000_0.2.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+-----------
+
+* [pr305](https://github.com/mono/Embeddinator-4000/pull/305) - [mtouch] Native symbols stripping now supported with Xamarin.iOS in embeddinator mode
+* [pr319](https://github.com/mono/Embeddinator-4000/pull/319) - [linker] Automatically create XML file to preserve anything required from SDK assemblies;
+
+macOS Support
+-------------
+
+* [pr324](https://github.com/mono/Embeddinator-4000/pull/324) - [mmp] Support for native frameworks generated with Xamarin.Mac;
+* [pr324](https://github.com/mono/Embeddinator-4000/pull/324) - [registrar] Support for subclassing managed subclasses of `NSObject` from Xamarin.Mac based assemblies;
```
Comment (rolfbjarne) [inline docs/releases/Embeddinator-4000_0.2.md]: Adding a link to the PR is not possible until the PR is created, which means we'll have to update the PR every time we create a PR...

Comment (rolfbjarne) [inline docs/releases/Embeddinator-4000_0.2.md]: Maybe we can add a milestone to every PR, then it should be possible to query for all the PRs for a milestone using a script and generate this list.

File: docs/releases/Embeddinator-4000_0.2.md (hunk 1 above)
Comment (spouliot) [inline docs/releases/Embeddinator-4000_0.2.md]: That's _hopefully_ not gonna happen very often going forward :)
but we could add a link to the `Make.config` until we're ready to release

File: docs/releases/Embeddinator-4000_0.2.md (hunk 2 above)
Comment (spouliot) [inline docs/releases/Embeddinator-4000_0.2.md]: good point, the PR # and links can come later (before release) - it's the description that matters the most (and blaming the file will give the details)

File: docs/releases/Embeddinator-4000_0.2.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+note: macOS support _without Xamarin.Mac_ remains supported but requires a compatible version of Mono to be installed on the computer;
+
+Known Issues
+============
+
+_to be updated_
+
+* This release only generates Objective-C code;
+* This release only targets macOS and iOS, the later requires Xamarin.iOS (10.11+) to be installed;
```
Comment (timrisi) [inline docs/releases/Embeddinator-4000_0.2.md]: the latter requires

File: docs/releases/Embeddinator-4000_0.2.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+* Some C# features (e.g. generics) are not yet supported;
+* In some cases the generator produce duplicate symbols, which won't compile;
```
Comment (timrisi) [inline docs/releases/Embeddinator-4000_0.2.md]: the generator produces

**Pull request merged**